Document Citation Support

dragonstyle opened the pull request:
- Add support for including `citation` in the front matter of documents. This allows you to specify the CSL-YAML that represents the citation information for this document (it will inherit the document metadata such as title when not specified). 

- Adds support for Google Scholar Metadata
Disabled by default, add `google-scholar` true to enable. Add additional metadata to configure properties. Generates references from bibliography.

- Adds support for citations in appendix
Includes a citation for the document in the appendix (bibtex and a formatted reference using the document CSL)

```yaml
---
title: "Test Document"
bibliography: references.bib
author: 
  - "Charles Teague"
  - J.J. Allaire
date: 1/23/2022
citation:
  type: article-journal
  container-title: The International Journal of Scientific Writing
  number: 7
  page: 12-15
  DOI: 10/223.2342432.2
google-scholar: true
---
```

Results in:
![Screen Shot 2022-03-01 at 2 11 19 PM](https://user-images.githubusercontent.com/261654/156233265-f93f0949-1cb3-4791-a94c-387cf096646f.png)







depends upon #361

dragonstyle committed:
Baseline implementation of scholar metadata

dragonstyle committed:
Improve attribute encoding

dragonstyle committed:
Baseline reference impl

dragonstyle committed:
Allow duplicate fields (needed for author, reference)

dragonstyle committed:
Add switch to disable google-scholar (default enabled)

dragonstyle committed:
Required this to be explicitely enabled for now

+ check for minimum requirements in order to include in Google Scholar

dragonstyle committed:
Add ‘Citation’ string

dragonstyle committed:
Basic bibliography rendering helper with CSL types and helpers

dragonstyle committed:
Localize appendix labels

jjallaire commented:
I think it would be fine for the `citation-url` to trigger the appendix (rather than `appendix-attribution` being required).

dragonstyle committed:
Cleanup

dragonstyle committed:
Automatically compute urls for document citations when possible

dragonstyle committed:
Make bibtex copyable

dragonstyle committed:
Add BibTeX and rendered citation to appendix, if called for

dragonstyle committed:
Remove `appendix-attribution`

dragonstyle committed:
Convert CC -> license, improve formatting

dragonstyle committed:
Simplify appendix styles

dragonstyle committed:
Move Google Scholar Rendering of Metadata into the core HTML format

dragonstyle committed:
Add a couple more csl fields

dragonstyle committed:
fix footnote spacing in appendix

dragonstyle committed:
Add suport for default type

We’ll look for publication or citation type, but ultimately we need some fallback type that can be selected by the format.

dragonstyle committed:
Cleanup unusued imports

dragonstyle committed:
Convert to citation namespace

dragonstyle committed:
Cleanup

dragonstyle committed:
Improve csl name documentation

dragonstyle committed:
Default appendix style

dragonstyle committed:
baseline csl schema

schema work

Correct schema

dragonstyle committed:
regen

dragonstyle committed:
remove disused console.og
